fix: added a few links and field numbering where order is involved (#40)
tari-project/rfcs: src/RFC-0121_ConsensusEncoding.md

The block hash pre-image is constructed by first constructing the merge mining hash. Each encoding is concatenated in order as follows:

1. `version` - [varint] 
2. `height` - [varint] 
3. `prev_hash` - [fixed sized array] 32 bytes 
4. `timestamp` - [varint] 
5. `input_mr` - [fixed sized array] 32 bytes
6. `output_mr` - [fixed sized array] 32 bytes
7. `output_mmr_size` - [varint] 
8. `witness_mr` - [fixed sized array] 32 bytes
9. `kernel_mr` - [fixed sized array] 32 bytes
10. `kernel_mmr_size` -  [varint]
11. `total_kernel_offset` - 32-byte Scalar, see [RistrettoPrivateKey]
12. `total_script_offset` - 32-byte Scalar, see [RistrettoPrivateKey]

This pre-image is hashed and block hash is constructed, in order, as follows:

1. `merge_mining_hash` - As above
2. `pow_algo` - enumeration of types of PoW as a single unsigned byte, where `Monero = 0x00` and `Sha3 = 0x01`

> 3. `pow_data` - raw variable bytes (no length varint)

Reviewer: ```suggestion
3. `pow_data` - raw variable bytes prepended with the length encoded as varint
```

}
```

[Output features](https://github.com/tari-project/tari/blob/development/base_layer/core/src/transactions/transaction_components/output_features.rs#L52) consensus encoding is defined as follows (in order):

1. `version` - 1 unsigned byte. This should always be `0x00` but is reserved for future proofing.

> 2. `maturity` - [varint]

Reviewer: The `1. 1. 1. ...` is markdown for a numbered list :)

```markdown
1. these 
1. are
1. all
1. 1s
```

Author: 1. numbers
2. like
3. the
4. count
5. :)

Reviewer 2: .. and now you have to manually update the numbers every time the list is edited..

The encoding is defined as follows:

> 1. `version` - [varint] 1 byte

Reviewer: ```suggestion
1. `version` - 1 byte
```

The transaction input canonical hash pre-image is constructed as follows:

> 1. `input_version` - [varint] 1 byte

Reviewer: ```suggestion
1. `version` - 1 byte
```

The transaction kernel is encoded as follows:

> 1. `input_version` - [varint] 1 byte

Reviewer: This isn't varint, just a single byte since we only use a u8 for input/output versions

https://github.com/tari-project/tari/blob/9162b4fa47cdc31e0cbbdfab0b8459eb5c58f7c5/base_layer/core/src/transactions/transaction_components/transaction_input_version.rs#L46

Reviewer: Oh it's the kernel, this is named `version` in the code
```suggestion
1. `version` - 1 byte
```
